UI/MapWnd.cpp

```diff
@@ -1915,7 +1915,7 @@ void MapWnd::RenderMovementLine(const MapWnd::MovementLineData& move_line, float
 
     // set vertex positions to outline a quad for each move line vertex
     for (std::vector<MovementLineData::Vertex>::const_iterator verts_it = vertices.begin();
-         verts_it != vertices.end(); ++verts_it)
+         verts_it != --vertices.end(); ++verts_it)
```

Comment:
I understand your motivation, but:
1) wouldn't that also cause problems if the vector is empty?
2) there is a test earlier in the function to make sure there are an odd number of entries in the vector...

https://github.com/matthoppe/freeorion/blob/null_pointer_fixes/UI/MapWnd.cpp#L1893